### PR TITLE
Fix to prevent appending or prepending self

### PR DESCRIPTION
**Pull request by SUPERCILEX**
If you somehow end up trying to append/prepend an item to itself, bad things happen. This PR blocks that.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -77,6 +80,9 @@ export class Item {
     if (!list) {
       return false
     }
+    if (this === item) {
```
Comment (wooorm) [inline index.js:83]: these need a blank line before them, and a comment explaining why this check is needed

**Comment by wooorm** [comment]
Also, tests!
And CI should pass :)

**Comment by SUPERCILEX** [comment]
Done!

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,6 @@
 .DS_Store
 *.log
+.idea
```
Comment (wooorm) [inline .gitignore:3]:
```suggestion
```

File: index.js [hunk 3]
```diff
@@ -32,6 +32,11 @@ export class Item {
       return false
     }
 
+    // Prevent broken pointers
+    if (this === item) {
+      return false
+    }
```
Comment (wooorm) [inline index.js:38]: Oh btw, better: move the two ifs-that-return-false into each other!

```
// If self is detached or prepending ourselves, return false.
if (!list || this === item) {
  return false
}
```

What do you think? Sorry I didn’t see this before!

**Review by wooorm — COMMENTED** [review]
Thanks!

File: index.js (hunk 3 above)
Comment (SUPERCILEX) [reply index.js:33]: Works for me!

File: .gitignore (hunk 2 above)
Comment (SUPERCILEX) [reply .gitignore:3]: See https://github.com/wooorm/linked-list/pull/21

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
thanks alex, [released](https://github.com/wooorm/linked-list/releases/tag/3.0.2)!

**Comment by SUPERCILEX** [comment]
Awesome, thanks!

**Pull request merged**